src/sentry/types/region.py

```diff
@@ -74,7 +74,18 @@ def to_url(self, path: str) -> str:
         """
         from sentry.api.utils import generate_region_url
 
-        return urljoin(generate_region_url(self.name), path)
+        if self.name == settings.SENTRY_MONOLITH_REGION:
+            base_url = options.get("system.url-prefix")
```

Comment:
This is likely the third time we've made this change back and forth.  I'm quite certain it used to be this.  I hope we converge on the correct solution.